web/pandas/pdeps/0005-no-default-index-mode.md

```diff
@@ -0,0 +1,377 @@
+accepted. For example, ``NoRowIndex`` would not necessarily need to subclass
+``RangeIndex``, and it would not necessarily need to be accessible to the user
+(``df.index`` could well return ``None``)
+
+## Likely FAQ
+
+**Q: Could not users just use ``RangeIndex``? Why do we need a new class?**
+
+**A**: ``RangeIndex`` is not preserved under slicing and appending, e.g.:
+  ```python
+  In [1]: ser = pd.Series([1,2,3])
+
+  In [2]: ser[ser!=2].index
+  Out[2]: Int64Index([0, 2], dtype='int64')
+  ```
+  If someone does not want to think about row labels and starts off
+  with a ``RangeIndex``, they'll very quickly lose it.
+
+**Q: Are indices not really powerful?**
+
+**A:** Yes! And they're also confusing to many users, even experienced developers.
+  It's fairly common to see pandas code with ``.reset_index`` scattered around every
+  other line. Such users would benefit from being able to not think about indices
+  and alignment. Indices would be here to stay, and ``NoRowIndex`` would not be the
+  default.
+
+**Q: How could one switch a ``NoRowIndex`` ``DataFrame`` back to one with an index?**
+
+**A:** The simplest way would probably be:
+  ```python
+  df.set_axis(pd.RangeIndex(len(df)))
+  ```
+  There's probably no need to introduce a new method for this.
+
+  Conversely, to get rid of the index, then (so long as one has enabled the ``mode.no_row_index`` option)
+  one could simply do ``df.reset_index(drop=True)``.
+
+**Q: How would ``tz_localize`` and other methods which operate on the index work on a ``NoRowIndex`` ``DataFrame``?**
+
+**A:** Same way they work on other ``NumericIndex``s, which would typically be to raise:
+
+  ```python
+  In [2]: ser.tz_localize('UTC')
+  ---------------------------------------------------------------------------
+  TypeError: index is not a valid DatetimeIndex or PeriodIndex
+  ```
+
+**Q: Why not let transpose switch ``NoRowIndex`` to ``RangeIndex`` under the hood before swapping index and columns?**
+
+**A:** This is the kind of magic that can lead to surprising behaviour that's
+  difficult to debug. For example, ``df.transpose().transpose()`` would not
+  round-trip. It's easy enough to set an index after all, better to "force" users
+  to be intentional about what they want and end up with fewer surprises later
+  on.
+
+**Q: What would df.sum(), and other methods which introduce an index, return?**
+
+**A:** Such methods would still set an index and would work the same way they
+  do now. There may be some way to change that (e.g. introducing ``as_index``
+  arguments and introducing a mode to set its default) but that's out of scope
+  for this particular PDEP.
+
+**Q: How would a user opt-in to a ``NoRowIndex`` DataFrame?**
+
+**A:** This PDEP would only allow it via the constructor, passing
+  ``index=NoRowIndex(len(df))``. A mode could be introduced to toggle
+  making that the default, but would be out-of-scope for the current PDEP.
```

Comment:
This is inconsistent with what you wrote above about having the `mode.no_row_index` option

Reply:
how so? even when I mention it above, I make it clear that this PDEP only deals with the `NoRowIndex` class, and that this mode would be out-of-scope for this PDEP

Reply:
There is text in the PDEP that refers to the mode that makes it seem like it is part of the proposal.  

For example, lines 297-298 read:
```text
Conversely, to get rid of the index, then (so long as one has enabled the ``mode.no_row_index`` option)
  one could simply do ``df.reset_index(drop=True)``.
```

When I read that, it seems like the mode is part of the proposal.   So I think you may need to qualify all references to `mode.no_row_index` to describe what would happen if the `mode.no_row_index` option is not available.

Or make a decision to include `mode.no_row_index` as part of the proposal.  (Which I think makes it easier to understand, IMHO)